minio/policy.py

```diff
@@ -240,8 +240,10 @@ def _remove_write_only(actions, cond_value):
 
     actions = _get_action(statement)
     cond_value = statement.get('Condition', {})
+    principal = statement.get('Principal')
     if (statement.get('Effect') == 'Allow' and
-            statement.get('Principal') == {'AWS': '*'}):
+            (principal == {'AWS': '*'} or
+            '*' in principal['AWS'])):
```

Comment:
This comparison is not very clean.  one check is dict match another one is string match.  We could avoid exact dict match.  You can do 
```py
aws = statement.get('Principal', {}).get("AWS", [])
```

then
```py
if (... and ('*' in aws or '*' in aws)):
```

Reply:
Done.